sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -710,39 +725,53 @@ synchronized void completeWorkflow() {
     /*
      * Store update token and possibly trigger application update check.
      */
-    synchronized void storeUpdateToken(@NonNull String updateToken, @NonNull String requestId) {
+    synchronized void storeRedirectionParameters(@NonNull String requestId, @NonNull String distributionGroupId, String updateToken) {
 
         /* Keep token for later if we are not started and enabled yet. */
         if (mContext == null) {
             MobileCenterLog.debug(LOG_TAG, "Update token received before onStart, keep it in memory.");
-            mBeforeStartUpdateToken = updateToken;
             mBeforeStartRequestId = requestId;
+            mBeforeStartUpdateToken = updateToken;
+            mBeforeStartDistributionGroupId = distributionGroupId;
         } else if (requestId.equals(PreferencesStorage.getString(PREFERENCE_KEY_REQUEST_ID))) {
-            String encryptedToken = CryptoUtils.getInstance(mContext).encrypt(updateToken);
-            PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_TOKEN, encryptedToken);
+            if (updateToken != null) {
+                String encryptedToken = CryptoUtils.getInstance(mContext).encrypt(updateToken);
+                PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_TOKEN, encryptedToken);
+            } else {
+                PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_TOKEN);
```

Comment:
It's not valid as of now as if we have something in storage we never open the browser again unless there is a fatal error which is another story...

Even if it was valid, to switch from private to public you would need to delete the token as keeping the token would make the SDK check for a private release.